Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/DeselectionHelper.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine.UIElements;
+
+namespace UnityEngine.InputSystem.Editor
+{
+    internal class DeselectionHelper
+    {
+        /// <summary>
+        /// Currently left as a constant to allow turning deselection on/off to be able to remove issues
+        /// related to not having selections more quickly. Also it might be desirable to either support or
+        /// not support deselection (E.g. pressing ESC while having a current selection).
+        /// </summary>
+        private const bool kAllowDeselection = false;
+
+        private int m_SelectedIndex = -1;
+
+        public bool Select(BaseVerticalCollectionView view, int selectedIndex)
```

Comment:
I agree with João. I think this method name is a bit confusing. Maybe something like IsSelectionValid()? As it's not really selecting or deselecting items in the method itself.

Reply:
I get that, I would like to avoid IsSelectionValid since that sounds like a read-only getter or evaluation method while this has side-effects for some cases and assigns selection to prevent deselection. Maybe ViewSelectionManager or similar is better? The function could be just be called Apply or be function operator overload if the class name makes sense since its basically a stateful functor.

Reply:
I have updated the code based on your reviews and generalised the selection filter so it would be ready to also handle multi-selection whenever/if we would like to support that (not in current development scope). Also renamed in an attempt to make class names and identifiers make more sense

Reply:
It's hard to find a simple lucid name here. I think your new changes are appropriate!

Reply:
Lets rename it later (its an internal) if a better name is found.